creator-node/src/middlewares/premiumContent/premiumContentMiddleware.ts

```diff
@@ -73,38 +79,51 @@ export const premiumContentMiddleware = async (
 
     switch (error) {
       case PremiumContentAccessError.MISSING_HEADERS:
-        return sendResponse(
+        prometheusResult.result = 'abort_premium_content_missing_headers'
+        return sendResponseWithMetric(
           req,
           res,
           errorResponseUnauthorized(
             'Missing request headers for premium content.'
-          )
+          ),
+          prometheusResult
         )
       case PremiumContentAccessError.INVALID_DISCOVERY_NODE:
-        return sendResponse(
+        prometheusResult.result =
+          'abort_premium_content_invalid_discovery_node_validation'
+        return sendResponseWithMetric(
           req,
           res,
           errorResponseForbidden(
             'Failed discovery node signature validation for premium content.'
-          )
+          ),
+          prometheusResult
         )
       case PremiumContentAccessError.FAILED_MATCH:
       default:
-        return sendResponse(
+        prometheusResult.result =
+          'abort_premium_content_failed_match_verification'
+        return sendResponseWithMetric(
           req,
           res,
           errorResponseForbidden(
             'Failed match verification for premium content.'
-          )
+          ),
+          prometheusResult
         )
     }
   } catch (e) {
-    const error = `Could not validate premium content access: ${
-      (e as Error).message
-    }`
     ;(req as any).logger.error(
```

Comment:
Are you able to change at `any` to `CustomRequest`

Reply:
oo i did not try. ill try in this pr review